ndk-build/src/apk.rs

```diff
@@ -94,6 +96,25 @@ impl<'a> UnalignedApk<'a> {
         Ok(())
     }
 
+    pub fn add_runtime_libs(
+        &self,
+        path: &Path,
+        target: Target,
+        search_paths: &[&Path],
+    ) -> Result<(), NdkError> {
+        let abi_dir = path.join(target.android_abi());
+        if abi_dir.is_dir() {
```

Comment:
an error message would be good if this condition fails

Reply:
Just printing a warning, or do you think I should return an error (such as `NdkError::PathNotFound`)?
That would stop the process. I'm not sure that's what we want here.

Reply:
It definitely needs to be an error since the user explicitly configures this path and should be told when it doesn't exist instead of silently going unnoticed (ie. typo, capitalization mismatch in the name).

You don't really need to do anything though - just remove `if abi_dir.is_dir()` and let `fs::read_dir(abi_dir)?` boil up the error if the path is invalid.

Though OTOH `io::Error` doesn't include the path leading to usual generic "file not found" message without any context. We should add a new error message expanding `Io(#[from] IoError)` with something like `IoPath(#[source] IoError, String)` and `.map_err(|io| IoError(io, path)`, but that's something for a different PR and to be applied globally! I can take care of this later unless you feel like adding it.

Reply:
I removed `if abi_dir.is_dir()` for now.
I don't mind making a PR for the error mapping at a later time if needed.

Reply:
In hindsight I don't see much if any applicable error unwraps (comment out `IoError` and follow all compiler errors for missing From implementations), feel free to add this for just this case in this PR.

Reply:
I'm not sure exactly how to do that, to be honest (still very much new to Rust), but I'll look into it.

Reply:
It's a fairly trivial thing to do. Instead of explaining I pushed another commit to this PR showing how :)

The error string isn't all too pretty but it's hard to provide a proper string when we have no idea what error occurred; just that we know it happened while interacting with the specified path.

Reply:
Thanks! 😀

Reply:
Yw! Any suggestions for a better string name though? And we could apply this to a few other locations but I'll reconsider where in a future PR.